Document grammar attributes

Kaiepi opened the pull request:
## The problem
Using attributes in grammars is undocumented.

## Solution provided
Document and provide an example using them.

<!--

    The template below contains optional suggestions. Simply omit it
    if you think it does not apply to this PR.

    Please state clearly in "The problem" what you are addressing with this
    pull request, referencing the issue(s) where it is described.

    In "Solution provided", tell us what you have done to address the
    problem.

-->

JJ commented on doc/Language/grammars.pod6:
```diff
@@ -288,7 +288,7 @@ the number for the currently parsed C<digit> digits.
     # OUTPUT: «(२५५ ४३५ ७७७)␤»
 
-=head2 Methods in Grammar
+=head2 Methods in Grammars
```
Can you please not use capitals here? Either you use C<Grammar>s or simply grammars.

Kaiepi commented:
Updated

Kaiepi commented:
I'll see if I can come up with a better example than this. I don't think this is very legible

JJ commented:
@Kaiepi the example is correct, but what you are explaining, attributes and methods, take a very small part of the total real estate. Maybe a simpler grammar would be enough...

Kaiepi commented:
Updated
I added more details since there was some stuff I was forgetting to mention, along with why attributes don't work in tokens. The example isn't much shorter, but should better illustrate how they behave
